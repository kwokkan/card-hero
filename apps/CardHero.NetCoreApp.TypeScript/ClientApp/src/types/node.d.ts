declare const module: {
    exports: {}
};

declare const process: {
    env: {
        NODE_ENV: "production" | "development"
    }
};
