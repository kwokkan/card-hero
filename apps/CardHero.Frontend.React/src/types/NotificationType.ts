export enum NotificationType {
    Default,

    Info,

    Success,

    Warning,

    Danger,
}
