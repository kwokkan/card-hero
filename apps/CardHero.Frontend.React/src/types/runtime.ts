export interface Action {
    (): void;
}

export interface Func<TResult> {
    (): TResult;
}
