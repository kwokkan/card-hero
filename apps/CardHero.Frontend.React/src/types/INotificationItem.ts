import { NotificationType } from "./NotificationType";

export interface INotificationItem {
    title?: string;
    message: string | JSX.Element;
    type?: NotificationType;
}
