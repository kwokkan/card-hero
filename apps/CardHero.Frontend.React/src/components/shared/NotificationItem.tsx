import React, { Component } from "react";
import { INotificationItem } from "../../types/INotificationItem";
import { NotificationType } from "../../types/NotificationType";

interface INotificationItemProps extends INotificationItem {
    onDismiss: () => void;
}

function typeToCss(type: NotificationType): string {
    switch (type) {
        case NotificationType.Danger:
            return "alert-danger";
        case NotificationType.Info:
            return "alert-info";
        case NotificationType.Success:
            return "alert-success";
        case NotificationType.Warning:
            return "alert-warning";
        default:
            return "alert-primary";
    }
}

export class NotificationItem extends Component<INotificationItemProps, {}> {
    render() {
        const typeClass = typeToCss(this.props.type);

        return (
            <div className={"notification-item alert alert-dismissible fade show " + typeClass}>
                {this.props.title &&
                    <h4 className="alert-heading">{this.props.title}</h4>
                }

                <p>{this.props.message}</p>

                <button type="button" className="close" data-dismiss="alert" aria-label="Close" onClick={() => this.props.onDismiss()}>
                    <span aria-hidden="true">&times;</span>
                </button>
            </div>
        );
    }
}
