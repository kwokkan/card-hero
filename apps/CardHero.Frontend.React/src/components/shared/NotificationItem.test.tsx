import React from "react";
import renderer from "react-test-renderer";
import { NotificationType } from "../../types/NotificationType";
import { NotificationItem } from "./NotificationItem";

test("<NotificationItem /> renders without title", () => {
    const tree = renderer
        .create(
            <NotificationItem message={"Test message"} onDismiss={null} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<NotificationItem /> renders with title and message", () => {
    const tree = renderer
        .create(
            <NotificationItem message={"Test message"} title={"Test title"} onDismiss={null} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<NotificationItem /> renders with title, message and custom type", () => {
    const tree = renderer
        .create(
            <NotificationItem message={"Test message"} title={"Test title"} type={NotificationType.Danger} onDismiss={null} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
