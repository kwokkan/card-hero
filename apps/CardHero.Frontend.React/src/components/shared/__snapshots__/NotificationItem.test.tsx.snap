// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationItem /> renders with title and message 1`] = `
<div
  className="notification-item alert alert-dismissible fade show alert-primary"
>
  <h4
    className="alert-heading"
  >
    Test title
  </h4>
  <p>
    Test message
  </p>
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;

exports[`<NotificationItem /> renders with title, message and custom type 1`] = `
<div
  className="notification-item alert alert-dismissible fade show alert-danger"
>
  <h4
    className="alert-heading"
  >
    Test title
  </h4>
  <p>
    Test message
  </p>
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;

exports[`<NotificationItem /> renders without title 1`] = `
<div
  className="notification-item alert alert-dismissible fade show alert-primary"
>
  <p>
    Test message
  </p>
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;
